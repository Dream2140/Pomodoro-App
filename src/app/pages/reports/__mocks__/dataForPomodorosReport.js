import { Helpers } from '../../../helpers/helpers';

export const dataForPomodorosReport = [
  {
    date: Helpers.getCurrentDate(),
    failed: 7,
    high: 1,
    low: 0,
    middle: 0,
    urgent: 3,
  },
];
