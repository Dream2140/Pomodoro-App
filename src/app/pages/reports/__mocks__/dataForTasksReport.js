import { Helpers } from '../../../helpers/helpers';

export const dataForTasksReport = [
  {
    date: Helpers.getCurrentDate(),
    failed: 1,
    high: 0,
    low: 0,
    middle: 0,
    urgent: 1,
  },
];
