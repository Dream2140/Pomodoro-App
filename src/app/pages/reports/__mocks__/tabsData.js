export const tabsData = {
    leftTabs: false,
    typeTabs: 'reports__top-tabs',
    tabPosition: 'tab__item--centered',
    tabData: [

      {
        period: true,
        val: 'day',
        data: 'Day',
        dataTab: 'data-reports-period="day"'
      },
      {
        period: true,
        val: 'week',
        data: 'Week',
        dataTab: 'data-reports-period="week"'
      },
      {
        period: true,
        val: 'month',
        data: 'Month',
        dataTab: 'data-reports-period="month"'
      }
    ]
};
