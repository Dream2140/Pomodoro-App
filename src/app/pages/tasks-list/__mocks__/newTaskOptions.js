export const newTaskOptions = {
  categoryId: 'other',
  completeDate: null,
  completedCount: [],

  deadline: 'April 5, 2022',
  
  description: 'Hello! Let`s start cleaning!',
  estimation: '5',
  failedPomodoros: [],
  priority: 'middle',
  status: 'GLOBAL_LIST',
  title: 'Test',
};
