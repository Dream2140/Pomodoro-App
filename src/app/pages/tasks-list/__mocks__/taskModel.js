export class TaskModel {
  constructor(options) {
    this.options = {
      title : options.title,
      description : options.description,
      failedPomodoros : options.failedPomodoros,
      completeDate : options.completeDate,
      completedCount : options.completedCount,
      priority : options.priority,
      categoryId : options.categoryId,
      status : options.status,
      estimation : options.estimation,
      deadline : options.deadline,
    }
  }
}