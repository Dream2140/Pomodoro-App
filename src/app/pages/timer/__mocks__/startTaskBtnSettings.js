export const startTaskBtnSettings = {
    secondBtn: true,
    firstColor: 'red',
    firstServiceClass: 'fail-pomodora',
    firstData: 'Fail Pomodora',
    secondColor: 'green',
    secondServiceClass: 'finish-pomodora',
    secondData: 'Finish Pomodora'
};