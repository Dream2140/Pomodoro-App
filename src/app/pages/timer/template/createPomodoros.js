export default function (amountOfPomodoros) {
    if (!amountOfPomodoros || !parseInt(amountOfPomodoros, 10)) {
 return ''; 
}
    let fragment = '';
  
    for (let i = 0; i < parseInt(amountOfPomodoros, 10); i++) {
      fragment += '<span class="estimation__item"></span>';
    }
  
    return fragment;
  }
  