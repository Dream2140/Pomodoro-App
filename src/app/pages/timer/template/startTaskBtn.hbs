<button class="service-btn__btn service-btn__btn--{{firstColor}} {{firstServiceClass}}">
    {{firstData}}
</button>

{{#if secondBtn}}
<button class="service-btn__btn service-btn__btn--{{secondColor}} {{secondServiceClass}}">
    {{secondData}}
</button>
{{/if}}