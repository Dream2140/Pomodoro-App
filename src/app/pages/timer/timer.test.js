/* eslint-disable no-undef */
import { TimerModel } from './scripts/timerModel';
import { tasks } from '../tasks-list/__mocks__/tasks';
import timerTemplate from './template/timer.hbs';
import startTaskBtn from './template/startTaskBtn.hbs';
import createPomodoros from './template/createPomodoros';
import { pomodoroSettings } from './__mocks__/pomodoroSettings';
import { activeTask } from './__mocks__/activeTask';
import { startTaskBtnSettings } from './__mocks__/startTaskBtnSettings';


const timerModel = new TimerModel();
timerModel.pomodoroSettings = pomodoroSettings;
timerModel.activeTask = activeTask;

describe('Test correct rendering', () => {

    it('renders timer template', () => {
        const tree = JSON.stringify(timerTemplate(tasks[0]));
        expect(tree).toMatchSnapshot();
    });

    it('renders start task button', () => {
        const tree = JSON.stringify(startTaskBtn(startTaskBtnSettings));
        expect(tree).toMatchSnapshot();
    });
});

describe('createPomodoros', () => {
    it('should create two pomodoros template', () => {
        const res = createPomodoros('2');
        const pomodorosTemplate = '<span class="estimation__item"></span><span class="estimation__item">';
        expect(res).toMatchSnapshot(pomodorosTemplate);
    });

    it('should return falsy value', () => {
        const res = createPomodoros();
        expect(res).toBeFalsy();
    });

    it('should return falsy value', () => {
        const res = createPomodoros('');
        expect(res).toBeFalsy();
    });
});

describe('Get data from storage', () => {
    beforeEach(() => {
        jest.spyOn(timerModel.settingsData, 'getDataFromStorage');
      });

    it('should get data from storage', () => {
        timerModel.getDataFromStorage('active-task');
        expect(timerModel.settingsData.getDataFromStorage).toHaveBeenCalledWith(
            'active-task'
        );
    });

    it('should return falsy value', () => {
        const res = timerModel.getDataFromStorage();
        expect(timerModel.settingsData.getDataFromStorage).not.toBeCalled();
        expect(res).toBeFalsy();
    });

    it('should return falsy value', () => {
        const res = timerModel.getDataFromStorage(23425);
        expect(timerModel.settingsData.getDataFromStorage).not.toBeCalled();
        expect(res).toBeFalsy();
    });
});