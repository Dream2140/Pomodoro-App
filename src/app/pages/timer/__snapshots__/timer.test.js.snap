// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test correct rendering renders start task button 1`] = `"\\"<button class=\\\\\\"service-btn__btn service-btn__btn--red fail-pomodora\\\\\\">\\\\r\\\\n    Fail Pomodora\\\\r\\\\n</button>\\\\r\\\\n\\\\r\\\\n<button class=\\\\\\"service-btn__btn service-btn__btn--green finish-pomodora\\\\\\">\\\\r\\\\n    Finish Pomodora\\\\r\\\\n</button>\\\\r\\\\n\\""`;

exports[`Test correct rendering renders timer template 1`] = `"\\" <header class=\\\\\\"header \\\\\\">\\\\r\\\\n     <div class=\\\\\\"wrapper\\\\\\">\\\\r\\\\n         <div class=\\\\\\"header__container\\\\\\">\\\\r\\\\n \\\\r\\\\n             <div class=\\\\\\"header__logo logo\\\\\\">\\\\r\\\\n                 <a href=\\\\\\"#\\\\\\" class=\\\\\\"logo__link\\\\\\">\\\\r\\\\n                     <img src=\\\\\\"/images/header.svg\\\\\\" alt=\\\\\\"header\\\\\\" class=\\\\\\"logo__img\\\\\\">\\\\r\\\\n                 </a>\\\\r\\\\n             </div>\\\\r\\\\n \\\\r\\\\n             <div class=\\\\\\"header__title-wrapp\\\\\\">\\\\r\\\\n                 <h1 class=\\\\\\"header__title\\\\\\">Daily task list</h1>\\\\r\\\\n             </div>\\\\r\\\\n \\\\r\\\\n             <nav class=\\\\\\"navigation header__nav\\\\\\">\\\\r\\\\n                 <ul class=\\\\\\"navigation__list\\\\\\">\\\\r\\\\n                     <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                         <button class=\\\\\\"navigation__btn \\\\\\" data-route=\\\\\\"task-list\\\\\\" title=\\\\\\"Go to Tasks\\\\\\">\\\\r\\\\n                             <span class=\\\\\\"navigation__icon icon icon-tasks\\\\\\"></span>\\\\r\\\\n                         </button>\\\\r\\\\n                     </li>\\\\r\\\\n                     <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                         <button class=\\\\\\"navigation__btn \\\\\\" data-route=\\\\\\"reports/day/tasks\\\\\\" title=\\\\\\"Go to Reports\\\\\\">\\\\r\\\\n                             <span class=\\\\\\"navigation__icon icon icon-statistics\\\\\\"></span>\\\\r\\\\n                         </button>\\\\r\\\\n                     </li>\\\\r\\\\n                     <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                         <button class=\\\\\\"navigation__btn \\\\\\"\\\\r\\\\n                             data-route=\\\\\\"settings/pomodoros\\\\\\" title=\\\\\\"Go to Settings\\\\\\">\\\\r\\\\n                             <span class=\\\\\\"navigation__icon icon  icon-settings\\\\\\"></span>\\\\r\\\\n                         </button>\\\\r\\\\n                     </li>\\\\r\\\\n                 </ul>\\\\r\\\\n             </nav>\\\\r\\\\n \\\\r\\\\n         </div>\\\\r\\\\n \\\\r\\\\n     </div>\\\\r\\\\n </header> <main class=\\\\\\"content timer-page\\\\\\">\\\\r\\\\n\\\\r\\\\n        <section class=\\\\\\"timer-section\\\\\\">\\\\r\\\\n            <div class=\\\\\\"wrapper\\\\\\">\\\\r\\\\n                <div class=\\\\\\"timer-section__container\\\\\\">\\\\r\\\\n                    <div class=\\\\\\"timer-section__info\\\\\\">\\\\r\\\\n                        <h1 class=\\\\\\"timer-section__title\\\\\\">Ывпыпа</h1>\\\\r\\\\n                        <p class=\\\\\\"timer-section_subtitle\\\\\\">Fgdgerf</p>\\\\r\\\\n                        <div class=\\\\\\"estimation timer-section__estimation\\\\\\">\\\\r\\\\n                            <div class=\\\\\\"estimation__container\\\\\\">\\\\r\\\\n                                <span class=\\\\\\"estimation__item\\\\\\"></span><span class=\\\\\\"estimation__item\\\\\\"></span><span class=\\\\\\"estimation__item\\\\\\"></span><span class=\\\\\\"estimation__item\\\\\\"></span><span class=\\\\\\"estimation__item\\\\\\"></span>\\\\r\\\\n                                <button class=\\\\\\"estimation__btn icon d-none icon-add\\\\\\"></button>\\\\r\\\\n                            </div>\\\\r\\\\n                        </div>\\\\r\\\\n                    </div>\\\\r\\\\n                    <div class=\\\\\\"timer-section__clock\\\\\\">\\\\r\\\\n                        <a href=\\\\\\"/#/task-list\\\\\\" class=\\\\\\"timer-section__link timer-section__link--left icon icon-arrow-left go-to-task-list\\\\\\" title=\\\\\\"Go to Global List\\\\\\"></a>\\\\r\\\\n                        <div class=\\\\\\"timer timer-section__timer\\\\\\" id=\\\\\\"timer\\\\\\">    \\\\r\\\\n\\\\r\\\\n                        </div>\\\\r\\\\n                        <a href=\\\\\\"/#/reports/day/tasks\\\\\\" class=\\\\\\"timer-section__link timer-section__link--right icon icon-right go-to-reports-page d-none\\\\\\" title=\\\\\\"Go to Report\\\\\\"></a>\\\\r\\\\n                    </div>\\\\r\\\\n                    <div class=\\\\\\"settings__buttons service-btn timer-section__btn\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"service-btn__btn service-btn__btn--green service-btn__btn-start-task\\\\\\">Start</button>\\\\r\\\\n                    </div>\\\\r\\\\n                </div>\\\\r\\\\n            </div>\\\\r\\\\n        </section>\\\\r\\\\n\\\\r\\\\n\\\\r\\\\n    </main>\\""`;

exports[`createPomodoros should create two pomodoros template: <span class="estimation__item"></span><span class="estimation__item"> 1`] = `"<span class=\\"estimation__item\\"></span><span class=\\"estimation__item\\"></span>"`;
